### PR TITLE
Test

### PR DESCRIPTION
**Pull request by OnyekachiSamuel**
Help me review my test

### CODE REVIEW

**Comment by rkterungwa16** [comment]
- Check for search terms with invalid characters such as 'what@#' should search for what
- check for words with uppercase letters 'whAt' should search for

File: tests/inverted-index-test.spec.js [hunk 1]
```diff
@@ -1,33 +1,67 @@
...
+      expect(invertedIndex.isMalformed(books)).toBe(false);
     });
 
-    it('should return true if the file is invalid', () => {
-      expect(invertedIndex.isInvalidFile(books)).toBe(true);
+    it('should return true if the file is valid', () => {
+      expect(invertedIndex.isValidFile(books)).toBe(true);
+    });
+
+    it('should return false if the file is invalid', () => {
```
Comment (andela-oawofadeju) [inline tests/inverted-index-test.spec.js]: You need to check this output and make adjustments. There some logical issues here.

File: tests/inverted-index-test.spec.js [hunk 2]
```diff
@@ -39,10 +73,11 @@
     it('should be the correct index', () => {
       expect(invertedIndex.createIndex('books.json', books)).toEqual(indexedFile);
```
Comment (andela-oawofadeju) [inline tests/inverted-index-test.spec.js]: You might want to pass in terms or an argument that would return the actual output like an index created and the position where the index is found within the object. This test is not efficient enough.

File: tests/inverted-index-test.spec.js [hunk 3]
```diff
@@ -39,10 +73,11 @@
...
     });
-  });
-
-
 
+    it('should be the correct index of the json file', () => {
+      expect(bookIndex).toEqual(index);
```
Comment (andela-oawofadeju) [inline tests/inverted-index-test.spec.js]: This test is not so efficient as well

File: tests/inverted-index-test.spec.js [hunk 4]
```diff
@@ -70,11 +105,38 @@
...
 
+    it('should handle and return correct result when the search terms is a mix of Capital and small letters', () => {
+      expect(invertedIndex.searchIndex(indexedFile, 'books.json', 'Wealth', ['Nations', 'This']))
+     .toEqual(
+       { wealth: [0], nations: [0], this: [0, 1] }
+     );
+    });
+
+    it('should handle and return correct result when the search terms is a mix of words and numbers', () => {
+      expect(invertedIndex.searchIndex(indexedFile, 'books.json', 'Wealth', ['Nations', 'This']))
```
Comment (andela-oawofadeju) [inline tests/inverted-index-test.spec.js]: Where are the mixed words and numbers indicated in the test case here???????? I'm not understanding?????

File: tests/inverted-index-test.spec.js [hunk 5]
```diff
@@ -70,11 +105,38 @@
...
+     .toEqual(
+       { wealth: [0], nations: [0], this: [0, 1] }
+     );
+    });
+
     it('should handle a varied number of search terms as arguments more thn', () => {
```
Comment (andela-oawofadeju) [inline tests/inverted-index-test.spec.js]: "more than"....what?????

File: tests/inverted-index-test.spec.js (hunk 4 above)
Comment (OnyekachiSamuel) [inline tests/inverted-index-test.spec.js]: The function filters out number input and returns only the valid inputs irrespective of whether they are lowercase or uppercase or a mixture of both.

File: tests/inverted-index-test.spec.js (hunk 5 above)
Comment (OnyekachiSamuel) [inline tests/inverted-index-test.spec.js]: I will correct that. Its omission.

File: tests/inverted-index-test.spec.js [hunk 6]
```diff
@@ -1,33 +1,67 @@
...
     });
 
-    it('should return true if the file is malformed', () => {
-      expect(invertedIndex.isMalformed(books)).toEqual('Valid');
+    it('should return false if the file is not malformed', () => {
+      expect(invertedIndex.isMalformed(malformed)).toEqual('Malformed');
+    });
+
+    it('should return true if the file is not malformed', () => {
+      expect(invertedIndex.isMalformed(books)).toBe(false);
```
Comment (OnyekachiSamuel) [inline tests/inverted-index-test.spec.js]: Thanks for that. I will correct that and revert.

File: tests/inverted-index-test.spec.js (hunk 1 above)
Comment (OnyekachiSamuel) [inline tests/inverted-index-test.spec.js]: I will check it out.

File: tests/inverted-index-test.spec.js (hunk 2 above)
Comment (OnyekachiSamuel) [inline tests/inverted-index-test.spec.js]: Noted. I will restructure the test.

File: tests/inverted-index-test.spec.js (hunk 3 above)
Comment (OnyekachiSamuel) [inline tests/inverted-index-test.spec.js]: Noted. I will restructure the test.

File: tests/inverted-index-test.spec.js (hunk 4 above)
Comment (andela-oawofadeju) [inline tests/inverted-index-test.spec.js]: So the input or terms indicated here do not have numbers attached to them let alone filtering. For example; input "WeAlTh01111" should return output "wealth". I can't see that here.

**Pull request merged**